src/radical/entk/pipeline/pipeline.py

```diff
@@ -9,15 +9,16 @@
 class Pipeline(object):
 
     """
-    A pipeline represents a collection of objects that have a linear temporal execution order.
-    In this case, a pipeline consists of multiple 'Stage' objects. Each ```Stage_i``` can execute only
-    after all stages up to ```Stage_(i-1)``` have completed execution.
+    A pipeline represents a collection of objects that have a linear temporal
+    execution order.  In this case, a pipeline consists of multiple 'Stage'
+    objects. Each ```Stage_i``` can execute only after all stages up to
+    ```Stage_(i-1)``` have completed execution.
 
     """
 
     def __init__(self):
 
-        self._uid = None
+        self._uid  = ru.generate_id('pipeline.%(item_counter)04d', ru.ID_CUSTOM)
```

Comment:
Isn't the namespace `(ns=sid)` needed?

Reply:
Test runs of `get_started.py` shows incremented uid indexes even though new session created.

- First run:
```
Update: pipeline.0000 state: SCHEDULING
Update: pipeline.0000.stage.0000 state: SCHEDULING
Update: pipeline.0000.stage.0000.my-first-task state: SCHEDULING
...
```

- Second run:
```
Update: pipeline.0001 state: SCHEDULING
Update: pipeline.0001.stage.0001 state: SCHEDULING
Update: pipeline.0001.stage.0001.my-first-task state: SCHEDULING
```

Reply:
Thanks, good catch!  the namespace is only really needed if the same type of entities get created in different components of the stack - which is not the case in RE.  But I indeed had to change from `item_counter` (which persists for the item type) to `counter` (which restarts for every application instance).  I made that consistent for all uid calls now.

First run:
```
re.session.rivendell.merzky.018351.0016
pipeline.0000 stage.0000 task.0000
pipeline.0000 stage.0001 task.0001
pipeline.0000 stage.0001 task.0002
```

Second run:
```
re.session.rivendell.merzky.018351.0017
pipeline.0000 stage.0000 task.0000
pipeline.0000 stage.0001 task.0001
pipeline.0000 stage.0001 task.0002
```